instant-xml-macros/src/lib.rs

```diff
@@ -22,15 +25,57 @@ fn retrieve_default_namespace(input: &DeriveInput) -> Option<String> {
         };
 
         if list.path.get_ident()? == "namespace" {
-            if let NestedMeta::Lit(Lit::Str(v)) = list.nested.first()? {
-                return Some(v.value());
-            }
+            return Some(list.nested.clone());
         }
     }
 
     None
 }
 
+fn retrieve_field_namespace(input: &syn::Field) -> Option<String> {
+    if let Some(list) = retrieve_namespace_list(&input.attrs) {
+        if let NestedMeta::Lit(Lit::Str(v)) = list.first()? {
+            return Some(v.value());
+        }
+    }
+    None
+}
+
+type DefaultNamespace = String;
+type OtherNamespaces = HashMap<String, String>;
+type Namespaces = Option<(Option<DefaultNamespace>, Option<OtherNamespaces>)>;
```

Comment:
Actually, I'd prefer not using type aliases, as they don't provide extra type safety, and usually just obscure what's actually going on. I also thinks it could make sense to just keep the empty `HashMap` in the case no other namespaces are defined, which should make the code handling the return value simpler.

Reply:
Agree, make sense, I simplified it, in my opinion, looks better now.